src/Microsoft.Health.Dicom.Core/Features/Store/StoreDatasetValidator.cs

```diff
@@ -218,6 +219,59 @@ string EnsureRequiredTagIsPresent(DicomTag dicomTag)
         StoreValidationResultBuilder validationResultBuilder)
     {
         IReadOnlyCollection<QueryTag> queryTags = await _queryTagService.GetQueryTagsAsync();
+
+        GenerateErrors(dicomDataset, validationResultBuilder, queryTags);
+
+        GenerateValidationWarnings(dicomDataset, validationResultBuilder, queryTags);
+    }
+
+    /// <summary>
+    ///  Generate errors on result by validating each core tag in dataset
+    /// </summary>
+    /// <remarks>
+    /// isCoreTag is utilized in store service when building a response and to know whether or not to create a failure or
+    /// success response. Anything added here results in a 409 conflict with errors in body.
+    /// </remarks>
+    /// <param name="dicomDataset"></param>
+    /// <param name="validationResultBuilder"></param>
+    /// <param name="queryTags"></param>
+    private void GenerateErrors(DicomDataset dicomDataset, StoreValidationResultBuilder validationResultBuilder,
+        IReadOnlyCollection<QueryTag> queryTags)
+    {
+        foreach (var requiredCoreTag in RequiredCoreTags)
+        {
+            try
+            {
+                // validate with additional leniency
+                var validationWarning =
+                    dicomDataset.ValidateDicomTag(requiredCoreTag, _minimumValidator, withLeniency: true);
```

Comment:
What is the reason for this new boolean? I see `ValidateDicomTag` called from 2 places, one is STOW and other in Re-index. We can have the same in re-indexing too right?